src/MessageModule/service/message.service.ts

```diff
@@ -1,47 +1,119 @@
-import {  
+import {
   Injectable,
-  InternalServerErrorException  
+  InternalServerErrorException,
+  NotFoundException,
+  NotAcceptableException,
 } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { MailerService } from '@nest-modules/mailer';
-import { ContactUsDTO, EmailDTO, SMSDTO } from '../dto';
+import { ContactUsDTO, EmailDTO } from '../dto';
+import { TemplateRepository } from '../repository/template.repository';
+import { Transactional } from 'typeorm-transactional-cls-hooked';
+import { TemplateDTO } from '../dto/templates.dto';
+import { TemplateMapper } from '../mapper/template.mapper';
 
 @Injectable()
 export class MessageService {
-
   constructor(
     private readonly mailerService: MailerService,
     private readonly configService: ConfigService,
-  ) {
-  }
-    
-  public async sendEmail(email: EmailDTO):Promise<void> {
+    private readonly templateRepository: TemplateRepository,
+    private readonly mapperTemplate: TemplateMapper,
+  ) { }
+
+  public async sendEmail(email: EmailDTO): Promise<void> {
     try {
       await this.mailerService.sendMail({
         to: this.configService.get<string>('EMAIL_CONTACTUS'),
         from: email.email,
         subject: email.title,
-        html: email.message        
+        html: email.message,
       });
     } catch (e) {
       throw new InternalServerErrorException(e);
     }
-  }  
+  }
 
-  public async sendContactUsEmail(contactUs: ContactUsDTO):Promise<void> {
+  public async sendContactUsEmail(contactUs: ContactUsDTO): Promise<void> {
     try {
       await this.mailerService.sendMail({
         to: this.configService.get<string>('EMAIL_CONTACTUS'),
         from: contactUs.email,
         subject: 'Fale conosco',
         template: 'contact-us',
         context: {
-          name: contactUs.name,          
-          message: contactUs.message
+          name: contactUs.name,
+          message: contactUs.message,
         },
       });
     } catch (e) {
       throw new InternalServerErrorException(e);
     }
-  }  
+  }
+
+  public async sendMessage(messageModel: any, templateName: string, contactEmail: string): Promise<void> {
+    try {
+      const model = await this.templateExists(templateName);
+      const message = this.transformObjectToArray(messageModel);
+      const htmlMessage = this.substituteInterpolationPerValues(model.template, message);
```

Comment:
Apenas para entendimento, qualquer objeto que for passado no messageModel ele procura pelo nome da propriedade e substitui o valor? 👌

Reply:
Isso, criei esse método para ser um coringa, onde passamos os valores que chegam do messageModel serão transformados em um array, na substituteInterpolationPerValues será substituído o que está no array sequencialmente na mensagem, por exemplo:
message = ['Filipe','Bom dia']
template = 'Olá {0}, {1}'
O que será retornado da função será: Olá Filipe, Bom dia.

Reply:
Eu faria como o @filipefelice falou, mas acho que esse modo que o kalil implementou não é ruim, acho que podemos manter assim. Caso a gente veja que trocar por nome invés de por sequencia seja melhor, mexemos depois